src/main/java/seedu/address/model/project/Repository.java

```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.project;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Project's repository.
+ */
+public class Repository {
+
+    private String projectRepository;
+
+    /**
+     * Construct's a project Repository.
+     *
+     * @param repository A valid repository name.
+     */
+    public Repository(String repository) {
+        requireNonNull(repository);
+        this.projectRepository = repository;
+    }
+
```

Comment:
Perhaps a simple method `getRepoUrl` could be implemented here, all Github repo links are of the form `github.com/USERNAME/REPO_NAME`, of which `USERNAME/REPO_NAME` is expected as input into `projectRepository`, hence a simple string concatenation could be carried out either here or in the `Parser` package. Regex string can also be matched here to ensure that the user input is of the form `USERNAME/REPO_NAME`

Reply:
I am aware that we can use regex string, but am unsure of what format aka what defines a username and repo-name as valid

Reply:
For the `USERNAME`, possible regex string can check for:
1. Maximum length of 39 characters
2. One or more alphanumeric characters (a-z, A-Z, 0-9)
3. Dashes (-)
4. Cannot begin with a dash
A possible regex string could be `"^[a-zA-Z0-9]([a-zA-Z0-9-]{0,38})"`

Regarding the `REPO_NAME`, a possible regex string could check for the following:
1. One or more alphanumeric characters (a-z, A-Z, 0-9)
2. Dashes (-)
3. Underscores (_)
4. Periods (.)

A possible regex string could be `"[a-zA-Z0-9-_.]+"`

The regex for these two would then be separated by a `/`.
Overall regex string is therefore `"^[a-zA-Z0-9]([a-zA-Z0-9-]{0,38})/[a-zA-Z0-9-_.]+"`.

I am sure there are much cleaner ways to create the regex string but this should suffice. Do remember to test the regex!
Syntax obtained from https://www.javatpoint.com/java-regex.

Reply:
I'll look into this later and see if I will make another commit